### DISCUSSION TITLE
What does MirrorMaker do when fails to publish an event?

### DISCUSSION THREAD

**Discussion by alizard0** [strimzi/strimzi-kafka-operator#6445] [Q&A] [answered]
Hello folks,

I have been reading the documentation and searching on stack overflow and could not find the answer for this question. 
When MirrorMaker fails to publish an event what is its fail-over process?

1. It does not commit the event.
2. It publishes the event on a DLQ
3. Ignores the event 

Does it work like the batch? https://stackoverflow.com/a/64222124/17932189

Many thanks!

**Comment by scholzj** [comment]
I never specifically tried this, but I would expect that it should normally not commit the consumed offset on the source and try it again.

**Comment by fvaleri** [comment, marked as answer]
MirrorMaker2 is built on top of KafkaConnect, which is built on top of KafkaClient APIs. This means that it works like you say, it simply try again. By default, producer's retries are MAX_INT. Note that the semantics is at_least_once, so your consumer on the target cluster should be idempotent if you don't want duplicates.
